src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -18,7 +18,10 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
             Debug.Assert(conversion.ConversionKind == ConversionKind.InterpolatedString);
             BoundExpression format;
             ArrayBuilder<BoundExpression> expressions;
-            MakeInterpolatedStringFormat((BoundInterpolatedString)conversion.Operand, out format, out expressions);
+            // TODO: Last argument could be true for some cases
+            // e.g. - ((FormattableString)$"Now: {DateTime.Now}").ToString(CultureInfo.GetCulture("some-lang"))
+            //      - FormattableString.Invariant($"Now: {DateTime.Now}")
```

Comment:
Let's just not deal with these.

Reply:
The second one is really bad as you pointed out.
Even the first one? I got it.

Reply:
@333fred How about these ones?
`FormattableString.Invariant` will not abuse this optimization and outside code cannot observe the optimization.

```cs
FormattableString.Invariant($"Name: {firstName} {lastName}")
// FormattableString.Invariant(FormattableStringFactory.Create("Name: {0} {1}", [firstName, lastName]))
// ↓
// string.Concat("Name: ", firstName, " ", lastName)

const string CompanyNameConst = "Your Company";
FormattableString.Invariant($"Copyright (c) {year} {CompanyNameConst}")
// FormattableString.Invariant(FormattableStringFactory.Create("Copyright (c) {0} {1}", [year, "Your Company"]))
// ↓
// FormattableString.Invariant(FormattableStringFactory.Create("Copyright (c) {0} Your Company", [year]))
```

Reply:
Let's just not deal with these.

Reply:
I've given up trying to optimize that case.